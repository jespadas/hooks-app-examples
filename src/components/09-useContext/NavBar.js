import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
    return (

        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">

            <Link to="/" className="navbar-brand">useContext</Link>

            <div className="navbar-collapse" id="navbarNav">
                <ul className="navbar-nav">


                    <NavLink
                        exact
                        activeClassName="active"
                        to="/"
                        className="nav-link"
                    >
                        Home
                    </NavLink>



                    <NavLink
                        exact
                        activeClassName="active"
                        to="/about"
                        className="nav-link"
                    >
                        About
                    </NavLink>

                    <NavLink
                        exact
                        activeClassName="active"
                        to="/login"
                        className="nav-link"
                    >
                        Login
                    </NavLink>


                </ul>
            </div>
        </nav>

    )
}
