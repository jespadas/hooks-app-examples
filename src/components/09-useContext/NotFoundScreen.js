import React from 'react';
import { Link } from 'react-router-dom';

export const NotFoundScreen = () => {
    return (
        <div>
            <h1> 404 - Not Found !</h1>
            <Link to="/">
                Go back home !
            </Link>
        </div>
    )
}
