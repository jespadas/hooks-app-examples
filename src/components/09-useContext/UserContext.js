import { createContext } from 'react';

/*
    useContext hook is used to create common data that can be accessed 
    throughout the component hierarchy without passing the props down 
    manually to each level. Context defined will be available to all 
    the child components without involving “props”.
*/

export const UserContext = createContext(null);
