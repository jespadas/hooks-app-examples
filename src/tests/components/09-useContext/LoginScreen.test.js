import React from 'react';
import { mount } from 'enzyme';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Tests in <LoginScreen />', () => {

    // Expected data
    const user = {
        id: 123,
        name: "Goku",
    };

    // Creating jest function
    const setUser = jest.fn();

    // Wrapping with mount to have acces to child
    const wrapper = mount(
        <UserContext.Provider value={{
            setUser: setUser
        }}>
            <LoginScreen />
        </UserContext.Provider>
    );

    test('Should be displayed correctly', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('Shoud execute setUser with same argument', () => {

        wrapper.find('button').prop('onClick')();

        expect(setUser).toHaveBeenCalledWith(user);

    });

});
