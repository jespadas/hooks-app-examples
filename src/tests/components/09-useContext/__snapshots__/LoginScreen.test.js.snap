// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <LoginScreen /> Should be displayed correctly 1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
